docs/source/json-api/production-setup.rst

```diff
@@ -0,0 +1,304 @@
+.. Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+Production Setup
+################
+
+The vast majority of the prior documentation focuses on ease of testing and running
+the service in a dev environment. From a production perspective given the wide
+variety of use-cases there is far less of an established framework for deploying
+the *JSON API*. In this document we would try to list some recommendations for
+production deployments.
+
+The *JSON API service* is a JVM application that by default uses an in-memory backend.
+This in-memory backend setup is inefficient for larger datasets as for every query it
+ends up fetching the entire active contract set for the templates referenced in that query.
+For this reason for production setups at a minimum we recommend to use a database
+as a query store, this will allow for more efficient caching of the data to improve
+query performance. Details for enabling a query store are highlighted below.
+
+Query Store
+***********
+
+Query store can be described as a cached search index and is useful for use-cases
+where we need to query large active contract sets. The *JSON API* can be configured
+with PostgreSQL/Oracle(Enterprise Edition only) as the query-store backend.
+
+For e.g to enable the PostgreSQL backend you can use the ``--query-store-jdbc-config`` flag, an example of which is below.
+
+.. note:: When you use the Query Store you'll want your first run to specify ``start-mode=create-only`` so that all the necessary tables are created. After the first run make sure ``start-mode=start-only`` so that it doesn't attempt to create the tables again.
+
+.. code-block:: shell
+
+    daml json-api --ledger-host localhost --ledger-port 6865 --http-port 7575 \
+    --query-store-jdbc-config "driver=org.postgresql.Driver,url=jdbc:postgresql://localhost:5432/test?&ssl=true,user=postgres,password=password,start-mode=start-only"
+
+
+If you are not familiar with JDBC URLs, we recommend reading the `PostgreSQL JDBC documentation <https://jdbc.postgresql.org/documentation/head/connect.html>`__
+for more information.
+
+
+Service Configuration
+*********************
+
+.. note:: The full list of configuration flags supported by the JSON API can be seen by running ``daml json-api --help``.
+
+Following are the list of recommended configs for a production deployment ? //TODO
+
+
+Security and privacy
+***************************
```

Comment:
```suggestion
Security and privacy
********************
```

Reply:
I believe there's a check in our docs build phase that ensures that header markups are the same length as the header text, sprinkled likewise suggestions all over the PR.